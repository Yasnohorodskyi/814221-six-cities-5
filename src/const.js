export const numberOfFavouritesOffers = 2;
export const AuthorizationCodes = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const SortType = {
  POPULAR: `Popular`,
  PRICE_LOW_TO_HIGH: `Price: low to high`,
  PRICE_HIGH_TO_LOW: `Price: high to low`,
  TOP_RATED_FIRST: `Top rated first`,
};
