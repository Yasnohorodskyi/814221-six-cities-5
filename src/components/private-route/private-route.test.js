import React from "react";
import renderer from "react-test-renderer";
import {PrivateRoute} from "./private-route";
import {AuthorizationCodes} from "../../const";
import {BrowserRouter as Router} from "react-router-dom";
const noop = () => {};
describe(`PrivateRoute /> render`, () => {
  it(`renders correctly with authorizationStatus = AUTH`, () => {
    const privateRouteComponent = renderer
      .create(
          <Router>
            <PrivateRoute
              exact={true}
              render={noop}
              authorizationStatus={AuthorizationCodes.AUTH}
              path={`/`}
            />
          </Router>
      )
      .toJSON();

    expect(privateRouteComponent).toMatchSnapshot();
  });

  it(`renders correctly with authorizationStatus = NO_AUTH`, () => {
    const privateRouteComponent = renderer
      .create(
          <Router>
            <PrivateRoute
              exact={true}
              render={noop}
              authorizationStatus={AuthorizationCodes.NO_AUTH}
              path={`/`}
            />
          </Router>
      )
      .toJSON();

    expect(privateRouteComponent).toMatchSnapshot();
  });
});
