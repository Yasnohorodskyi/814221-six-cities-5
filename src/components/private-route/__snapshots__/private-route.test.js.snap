// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute /> render renders correctly with authorizationStatus = AUTH 1`] = `null`;

exports[`PrivateRoute /> render renders correctly with authorizationStatus = NO_AUTH 1`] = `null`;
